src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph_RemoveProject.cs

```diff
@@ -90,31 +97,29 @@ private static ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> Compu
     private static ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>>? ComputeNewReverseReferencesMapForRemovedProject(
         ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> existingForwardReferencesMap,
         ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>>? existingReverseReferencesMap,
-        ProjectId removedProjectId)
+        ArrayBuilder<ProjectId> removedProjectIds)
     {
+        // If the map was never calculated for the previous graph, so there is nothing to update.
         if (existingReverseReferencesMap is null)
-        {
-            // The map was never calculated for the previous graph, so there is nothing to update.
             return null;
-        }
-
-        if (!existingForwardReferencesMap.TryGetValue(removedProjectId, out var forwardReferences))
-        {
-            // The removed project did not reference any other projects, so we simply remove it.
-            return existingReverseReferencesMap.Remove(removedProjectId);
-        }
 
         var builder = existingReverseReferencesMap.ToBuilder();
-
-        // Iterate over each project referenced by 'removedProjectId', which is now being removed. Update the
-        // reverse references map for the project to no longer include 'removedProjectId' in the list.
-        foreach (var referencedProjectId in forwardReferences)
+        foreach (var removedProjectId in removedProjectIds)
         {
-            builder.MultiRemove(referencedProjectId, removedProjectId);
+            // If the removed project did not reference any other projects, nothing to do for this project.
+            if (!existingForwardReferencesMap.TryGetValue(removedProjectId, out var forwardReferences))
+                continue;
+
+            // Iterate over each project referenced by 'removedProjectId', which is now being removed. Update the
+            // reverse references map for the project to no longer include 'removedProjectId' in the list.
+            foreach (var referencedProjectId in forwardReferences)
+                builder.MultiRemove(referencedProjectId, removedProjectId);
         }
 
         // Finally, remove 'removedProjectId' itself.
-        builder.Remove(removedProjectId);
+        foreach (var removedProjectId in removedProjectIds)
```

Comment:
RemoveRange; ditto for the rest of the file.

Reply:
See earlier comment. I don't feel *super* opinionated, so if you are, I can go ahead and change them.

Reply:
Closing this in favor of prior comment being where we discuss.